airflow/api_connexion/endpoints/task_instance_endpoint.py

```diff
@@ -286,9 +287,13 @@ def post_set_task_instances_state(dag_id, session):
     except ValidationError as err:
         raise BadRequest(detail=str(err.messages))
 
-    dag = current_app.dag_bag.get_dag(dag_id)
-    if not dag:
-        error_message = "Dag ID {} not found".format(dag_id)
+    error_message = "Dag ID {} not found".format(dag_id)
+    try:
+        dag = current_app.dag_bag.get_dag(dag_id)
+        if not dag:
+            raise NotFound(error_message)
+    except SerializedDagNotFound:
```

Comment:
If we made AirflowNotFound inherit from connextion.NotFound this code could be a simpler here.

WDYT?

Reply:
I will create a separate issue for it, to address it in a separate PR (not strictly related to this PR), so anyone else can also pick that task too.